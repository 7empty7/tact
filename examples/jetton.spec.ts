import { toNano } from "ton-core";
import { ContractSystem, testAddress } from "ton-emulator";
import { SampleJetton } from './output/jetton_SampleJetton';

describe('jetton', () => {
    it('should deploy', async () => {

        // Create jetton
        let system = await ContractSystem.create();
        let owner = system.treasure('owner');
        let contract = system.open(await SampleJetton.fromInit(owner.address, null));

        // Mint
        await contract.send(owner, { value: toNano(1) }, { $$type: 'Mint', amount: toNano(1000000) });
        await system.run();

        // Check owner
        expect((await contract.getOwner()).toString()).toEqual(owner.address.toString());

        // Data
        let data = await contract.getGetJettonData();
        // console.warn(data);
    });
});