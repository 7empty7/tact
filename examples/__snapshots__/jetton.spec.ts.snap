// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jetton should deploy 1`] = `
[
  {
    "type": "deploy",
  },
  {
    "message": {
      "body": {
        "cell": "x{01FB345B0000000000000000000000000000000000000000000000000001C6BF526340004_}",
        "type": "cell",
      },
      "bounce": true,
      "from": "kQAI-3FJVc_ywSuY4vq0bYrzR7S4Och4y7bTU_i5yLOB3A6P",
      "to": "kQDjZfqbMtoRCYt763mzVMiZZ8NoXOchBGH2Njn1Hn78gfSB",
      "type": "internal",
      "value": 1000000000n,
    },
    "type": "received",
  },
  {
    "gasUsed": 15992n,
    "type": "processed",
  },
  {
    "messages": [
      {
        "body": {
          "cell": "x{178D451900000000000000007038D7EA4C68000801C6CBF53665B4221316F7D6F366A99132CF86D0B9CE4208C3EC6C73EA3CFDF90300023EDC525573FCB04AE638BEAD1B62BCD1ED2E0E721E32EDB4D4FE2E722CE07702_}",
          "type": "cell",
        },
        "bounce": false,
        "from": "kQDjZfqbMtoRCYt763mzVMiZZ8NoXOchBGH2Njn1Hn78gfSB",
        "to": "kQCPC3MRKf5IPXJYB8wvUZu-ruo3Bj_ztD_pNSlO26FLfxCj",
        "type": "internal",
        "value": 967641000n,
      },
    ],
    "type": "sent",
  },
]
`;
